hs_core/models.py

```diff
@@ -2025,26 +2523,20 @@ def get_crossref_deposit_xml(self, pretty_print=True):
 
     @property
     def size(self):
+        """
+        Return the total size of all data files in iRODS.
 
-        total_file_size = 0
-
+        This size does not include metadata. Just files. Specifically,
+        resourcemetadata.xml, systemmetadata.xml are not included in this
+        size estimate.
+        """
         try:
             # compute the total file size for the resource
-            f_sizes = [f.resource_file.size
-                       if f.resource_file else 0
-                       for f in self.files.all()]
-            total_file_size = sum(f_sizes)
-
-            # compute the total file size for federated resource
-            f_sizes = [int(f.fed_resource_file_size)
-                       if f.fed_resource_file_size else 0
-                       for f in self.files.all()]
-            total_file_size += sum(f_sizes)
+            f_sizes = [f.size for f in self.files.all()]
+            return sum(f_sizes)
 
         except SessionException:
-            pass
-
-        return total_file_size
+            return -1
```

Comment:
What should I do in the case where I can't read the size? I only have three options: 
1. Raise an exception. 
2. Return a flag value
3. Return zero. 
Of these, only (1) makes any real sense.